src/dotnet/commands/dotnet-compile-csc/Program.cs

```diff
@@ -84,6 +84,7 @@ public static int Run(string[] args)
 
                 // Execute CSC!
                 var result = RunCsc(new string[] { $"-noconfig", "@" + $"{rsp}" })
+                    .WorkingDirectory(Directory.GetCurrentDirectory())
```

Comment:
There is an existing Command class that only knows how to spin up processes:

https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/Command.cs

It and BuiltInCommand have very little common implementation.

WorkingDirectory is already on ICommand:  https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/ICommand.cs#L13.  

The two classes only share the interface.